share/spice/seat_geek/events_by_artist/seat_geek_events_by_artist.js

```diff
@@ -0,0 +1,168 @@
+(function(env) {
+    "use strict";
+    env.ddg_spice_seat_geek_events_by_artist = function(api_result) {
+
+        if(!api_result || api_result.error || api_result.events.length === 0) {
+            return Spice.failed('seat_geek_events_by_artist');
+        }
+
+         // Get the original query.
+        var script = $('[src*="/js/spice/seat_geek/events_by_artist/"]')[0],
+            source = $(script).attr("src"),
+            artistName = source.match(/events_by_artist\/([^\/]*)/)[1];
+
+        var months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'June', 'July', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec'],
+            days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
+            events;
+            
+        // sometimes we get false positives from the SeatGeek API
+        // where for example sports games get displayed instead of concerts
+        //
+        // filtering out events whose performers don't have genres
+        // seems to be a way to guard against that
+        events = $.grep(api_result.events, function (item) {
```

Comment:
@andrey-p Could you please move the check inside the normalize function? That goes already through all the items one at a time, so you can just check whether at least one of the performers for that event has the "genre" key, and if that's not the case, return `null`, so that item will be skipped.